components/central-application-gateway/internal/proxy/reverseproxy.go

```diff
@@ -97,6 +129,14 @@ func responseModifier(
 	urlRewriter func(gatewayURL, target, loc *url.URL) *url.URL,
 ) func(*http.Response) error {
 	return func(resp *http.Response) error {
+		log := zap.L().Sugar()
+		_ = httptools.LogResponse(log, resp)
+
+		if resp.StatusCode >= 500 && resp.StatusCode < 600 {
+			resp.Header.Set("Target-System-Status", strconv.Itoa(resp.StatusCode))
+			resp.StatusCode = http.StatusBadGateway
```

Comment:
I have no knowledge to determine if setting `502` status here is or is not a good way of handling that. 
Would be good to get someone's else opinion.

Reply:
Also, this change should be in my opinion created as a separate PR